libcaf_core/test/detail/rfc3629.cpp

```diff
@@ -122,43 +123,107 @@ constexpr std::string_view ascii_2 = R"__(
  *                      \____/_/   \_|_|                                      *
 )__";
 
-} // namespace
-
-TEST_CASE("ASCII input") {
-  CHECK(valid_utf8(ascii_1));
-  CHECK(valid_utf8(ascii_2));
+TEST("rfc3629::valid checks whether an input is valid UTF-8") {
+  SECTION("valid ASCII input") {
+    check(valid_utf8(ascii_1));
+    check(valid_utf8(ascii_2));
+  }
+
+  SECTION("valid UTF-8 input") {
+    check(valid_utf8(valid_two_byte_1));
+    check(valid_utf8(valid_two_byte_2));
+    check(valid_utf8(valid_three_byte_1));
+    check(valid_utf8(valid_three_byte_2));
+    check(valid_utf8(valid_four_byte_1));
+    check(valid_utf8(valid_four_byte_2));
+  }
+
+  SECTION("invalid UTF-8 input") {
+    check(!valid_utf8(invalid_two_byte_1));
+    check(!valid_utf8(invalid_two_byte_2));
+    check(!valid_utf8(invalid_two_byte_3));
+    check(!valid_utf8(invalid_two_byte_4));
+    check(!valid_utf8(invalid_three_byte_1));
+    check(!valid_utf8(invalid_three_byte_2));
+    check(!valid_utf8(invalid_three_byte_3));
+    check(!valid_utf8(invalid_three_byte_4));
+    check(!valid_utf8(invalid_three_byte_5));
+    check(!valid_utf8(invalid_three_byte_6));
+    check(!valid_utf8(invalid_three_byte_7));
+    check(!valid_utf8(invalid_three_byte_8));
+    check(!valid_utf8(invalid_four_byte_1));
+    check(!valid_utf8(invalid_four_byte_2));
+    check(!valid_utf8(invalid_four_byte_3));
+    check(!valid_utf8(invalid_four_byte_4));
+    check(!valid_utf8(invalid_four_byte_5));
+    check(!valid_utf8(invalid_four_byte_6));
+    check(!valid_utf8(invalid_four_byte_7));
+    check(!valid_utf8(invalid_four_byte_8));
+    check(!valid_utf8(invalid_four_byte_9));
+    check(!valid_utf8(invalid_four_byte_10));
+  }
 }
 
-TEST_CASE("valid UTF-8 input") {
-  CHECK(valid_utf8(valid_two_byte_1));
-  CHECK(valid_utf8(valid_two_byte_2));
-  CHECK(valid_utf8(valid_three_byte_1));
-  CHECK(valid_utf8(valid_three_byte_2));
-  CHECK(valid_utf8(valid_four_byte_1));
-  CHECK(valid_utf8(valid_four_byte_2));
+TEST("rfc3629::validate returns the end index if the range is valid") {
+  SECTION("valid ASCII input") {
+    check_eq(rfc3629::validate(ascii_1), res_t{ascii_1.size(), false});
+    check_eq(rfc3629::validate(ascii_2), res_t{ascii_2.size(), false});
+  }
+
+  SECTION("valid UTF-8 input") {
+    check_eq(rfc3629::validate(valid_two_byte_1), res_t{2, false});
```

Comment:
I'd like to avoid hard-coding/guessing the sizes here. :)

I think we could use either `std::size()` here or change the types of these things to `std::array`.

Reply:
I taught the `two` in `valid_two_byte_1` was enough to indicate the size, but sure, I can use size. The downside is it might break the line in two.

Reply:
Yeah, you're right. I think I didn't read the variable name. Keep it that way. 👍